doc/source/rllib/package_ref/replay_buffers.rst

```diff
@@ -0,0 +1,132 @@
+
+        .. automethod:: __init__
+            :noindex:
+        .. automethod:: add
+            :noindex:
+        .. automethod:: sample
+            :noindex:
+
+
+Building your own ReplayBuffer
+------------------------------
+Here is an example of how to implement your own toy example of a ReplayBuffer class and make SimpleQ use it:
+
+.. literalinclude:: ../../../../rllib/examples/documentation/replay_buffer_demo.py
+    :language: python
+    :start-after: __sphinx_doc_replay_buffer_own_buffer__begin__
+    :end-before: __sphinx_doc_replay_buffer_own_buffer__end__
+
+For a full implementation, you should consider other methods like get_state() and set_state() methods.
+A more extensive example is our implementation of `reservoir sampling <https://www.cs.umd.edu/~samir/498/vitter.pdf>`__:
+
+.. dropdown:: **Changing a replay buffer configuration**
+    :animate: fade-in-slide-down
+
+    .. literalinclude:: ../../../../rllib/utils/replay_buffers/reservoir_replay_buffer.py
+        :language: python
+        :start-after: __sphinx_doc_reservoir_buffer__begin__
+        :end-before: __sphinx_doc_reservoir_buffer__end__
+
+
+Advanced Usage
+==============
+
+In RLlib, all replay buffers implement the :py:class:`~ray.rllib.utils.replay_buffers.replay_buffer.ReplayBuffer` interface.
+Therefore, they support, whenever possible, different :py:class:`~ray.rllib.utils.replay_buffers.replay_buffer.StorageUnit`\s.
+The storage_unit constructor argument of a replay buffer defines how experiences are stored, and therefore the unit in which they are sampled.
+When later calling the sample() method, num_items will relate to said storage_unit.
+
+Here is a full example of how to modify the storage_unit and interact with a custom buffer:
+
+.. literalinclude:: ../../../../rllib/examples/documentation/replay_buffer_demo.py
+    :language: python
+    :start-after: __sphinx_doc_replay_buffer_advanced_usage_storage_unit__begin__
+    :end-before: __sphinx_doc_replay_buffer_advanced_usage_storage_unit__end__
+
+As noted above, Rllib's :py:class:`~ray.rllib.utils.replay_buffers.multi_agent_replay_buffer.MultiAgentReplayBuffer`\s
+support modification of underlying replay buffers. Under the hood, the :py:class:`~ray.rllib.utils.replay_buffers.multi_agent_replay_buffer.MultiAgentReplayBuffer`
+stores experiences per policy in separate underlying replay buffers. You can modify their behaviour by specifying an underlying ``replay_buffer_config`` that works
+the same way as the parent's config.
+
+Here is an example of how to specify an underlying replay buffer along with a default call argument:
+
+.. literalinclude:: ../../../../rllib/examples/documentation/replay_buffer_demo.py
+    :language: python
+    :start-after: __sphinx_doc_replay_buffer_advanced_usage_underlying_buffers__begin__
+    :end-before: __sphinx_doc_replay_buffer_advanced_usage_underlying_buffers__end__
+
+
+Replay Buffers API Reference
+============================
+
+.. toctree::
```

Comment:
I don't like that this is embedded as TOC here. You have to click on these, which brings you to separate pages that are not contained in the main navigation (how to get back?).